src/widgets/modalChildren/ADRInput.js

```diff
@@ -1,30 +1,90 @@
 /* eslint-disable react/forbid-prop-types */
-import React, { useState } from 'react';
-import { StyleSheet } from 'react-native';
+import React, { useEffect, useState, useLayoutEffect } from 'react';
+import { ActivityIndicator, StyleSheet, Text } from 'react-native';
 import { connect } from 'react-redux';
 import PropTypes from 'prop-types';
 import { JSONForm } from '../JSONForm/JSONForm';
 import { UIDatabase } from '../../database/index';
 import { FlexRow } from '../FlexRow';
+import { FlexView } from '../FlexView';
 import { PageButton } from '../PageButton';
-import { SUSSOL_ORANGE } from '../../globalStyles/colors';
+import { SUSSOL_ORANGE, WHITE } from '../../globalStyles/colors';
 import globalStyles from '../../globalStyles/index';
 import { PatientActions } from '../../actions/PatientActions';
 import { selectCurrentPatient } from '../../selectors/patient';
+import { useLocalAndRemotePatientHistory } from '../../hooks/useLocalAndRemoteHistory';
+import { dispensingStrings } from '../../localization';
 
-export const ADRInputComponent = ({ onCancel, onSave, patient }) => {
+const getSchemaItems = jsonSchema => {
+  const { properties = {} } = jsonSchema;
+  const { causes = {} } = properties;
+  const { items } = causes;
+  return items;
+};
+const mapHistory = history =>
+  history.map((h, index) => {
+    const { prescriberOrVaccinator, itemName, confirmDate } = h;
+    const vaccinator = prescriberOrVaccinator ? `  Vaccinator: ${prescriberOrVaccinator}` : '';
+
+    return `${index + 1}. ${itemName}  Date: ${confirmDate}${vaccinator}`;
+  });
+
+const LoadingIndicator = ({ loading }) => (
+  <FlexView flex={1} justifyContent="center" alignItems="center" style={{ marginTop: 20 }}>
+    {loading && (
+      <>
+        <Text style={localStyles.text}>{dispensingStrings.fetching_history}</Text>
+        <ActivityIndicator color={SUSSOL_ORANGE} size="small" style={{ marginTop: 10 }} />
+      </>
+    )}
+  </FlexView>
+);
+
+LoadingIndicator.propTypes = {
+  loading: PropTypes.bool.isRequired,
+};
+
+export const ADRInputComponent = ({ onCancel, onSave, patient, patientHistory }) => {
   const [{ formData, isValid }, setForm] = useState({ formData: null, isValid: false });
+  const patientId = patient?.id;
+  const [ADRSchema, setADRSchema] = useState(UIDatabase.objects('ADRForm')[0]);
+  const { jsonSchema, uiSchema, type, version } = UIDatabase.objects('ADRForm')[0];
+  const items = getSchemaItems(jsonSchema);
+  const [{ data, loading, searched }, fetchOnline] = useLocalAndRemotePatientHistory({
+    isVaccine: true,
```

Comment:
Oh no this is my bad, I renamed `isVaccine` to `isVaccineDispensingModal` without updating the reference in the `useLocalAndRemoteHistory` hook so now non-vaccines are showing up in vaccine modals.... 😿 : https://github.com/openmsupply/mobile/blob/6a58b9377b7d50a7c3444033ebc8ae53de484797/src/sync/lookupApiUtils.js#L206

...soz

Reply:
thanks - I missed that!
6d16feed4